homeassistant/components/indevolt/sensor.py

```diff
@@ -0,0 +1,289 @@
+"""Sensor platform for Indevolt."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Final, cast
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory, UnitOfEnergy, UnitOfPower
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from .const import DOMAIN
+from .entity import IndevoltEntity
+
+
+PARALLEL_UPDATES = 1
+
+WORKING_MODE = {
+    0: "Outdoor Portable",
```

Comment:
Enum sensor state value strings should be lowercase snake_case. Use translation strings for nicer formatting.